app/scripts/metamask-controller.js

```diff
@@ -1069,7 +1070,6 @@ export default class MetamaskController extends EventEmitter {
         snapKeyringBuildMessenger,
         getSnapController,
         persistAndUpdateAccounts,
-        (address) => this.preferencesController.setSelectedAddress(address),
```

Comment:
This is no longer needed. We set the selected address using the controller messenger.